### ISSUE TITLE
Element.cloneNode() constructs HTMLUnknownElements

### ISSUE THREAD

**Issue by westonruter** [ampproject/worker-dom#614]
Consider this code:

```html
  <amp-script script="clone-test">
    <button type=button id=clone>
		Clone!
    </button>
    <p>
      This <strong>element</strong> <u>will</u> be <em>cloned</em>.
    </p>
  </amp-script>
  <script id="clone-test" type="text/plain" target="amp-script">
	document.getElementById( 'clone' ).addEventListener( 'click', () => {
    	const cloned = document.querySelector('p').cloneNode( true );
        document.body.appendChild( cloned );
    } );
  </script>
```

Upon clicking the Clone button which duplicates the paragraph, the resulting DOM rendered is:

> ![image](https://user-images.githubusercontent.com/134745/63118713-486ffb80-bf53-11e9-8eba-32e3724a040d.png)

Notice an unexpected loss of style for the elements.  Investigating in DevTools shows unexpectedly that the cloned elements have upper-case names:

> ![image](https://user-images.githubusercontent.com/134745/63118786-705f5f00-bf53-11e9-9eee-62f523aef215.png)

This is likely a side-effect of the fact that the cloned elements are not recognized HTML elements. They are instances of `HTMLUnknownElement`:

> ![image](https://user-images.githubusercontent.com/134745/63118908-b3213700-bf53-11e9-9a6e-4bb69c409460.png)

**Comment by westonruter** [comment]
This is a workaround `cloneNode` implementation that is working in my use case:

```js
/**
 * Clone node.
 *
 * This is a workaround where Element.cloneNode() seems to fail on some elements (e.g. tr and td).
 *
 * @param {Node|Element} node - Node
 * @param {Boolean} deep - Deep cloning.
 * @returns {Node} Cloned node.
 */
export const cloneNode = ( node, deep = false ) => {
    if ( 1 !== node.nodeType ) {
        return node.cloneNode( deep );
    }

    const cloned = document.createElement( node.nodeName );
    for ( let i = 0; i < node.attributes.length; i++ ) {
        cloned.setAttribute( node.attributes[i].name, node.attributes[i].value );
    }

    if ( deep ) {
        cloned.innerHTML = node.innerHTML;
    }

    return cloned;
};
```

**Comment by kristoferbaxter** [comment]
Found the fix for this, but want to add a few more tests. Will submit tomorrow.

**Comment by kristoferbaxter** [comment]
Released with `0.18.0`.
